rct229/rulesets/ashrae9012019/ruleset_functions/get_zone_BPF_BAT.py

```diff
@@ -0,0 +1,144 @@
+    LIGHTING_SPACE.GYMNASIUM_FITNESS_CENTER_PLAYING_AREA: "UNDETERMINED",
+    LIGHTING_SPACE.HEALTHCARE_FACILITY_EMERGENCY_ROOM: "HEALTHCARE_HOSPITAL",
+    LIGHTING_SPACE.HEALTHCARE_FACILITY_EXAM_TREATMENT_ROOM: "HEALTHCARE_HOSPITAL",
+    LIGHTING_SPACE.HEALTHCARE_FACILITY_MEDICAL_SUPPLY_ROOM: "HEALTHCARE_HOSPITAL",
+    LIGHTING_SPACE.HEALTHCARE_FACILITY_NURSERY: "HEALTHCARE_HOSPITAL",
+    LIGHTING_SPACE.HEALTHCARE_FACILITY_NURSES_STATION: "HEALTHCARE_HOSPITAL",
+    LIGHTING_SPACE.HEALTHCARE_FACILITY_OPERATING_ROOM: "HEALTHCARE_HOSPITAL",
+    LIGHTING_SPACE.HEALTHCARE_FACILITY_PATIENT_ROOM: "HEALTHCARE_HOSPITAL",
+    LIGHTING_SPACE.HEALTHCARE_FACILITY_PHYSICAL_THERAPY_ROOM: "HEALTHCARE_HOSPITAL",
+    LIGHTING_SPACE.HEALTHCARE_FACILITY_RECOVERY_ROOM: "HEALTHCARE_HOSPITAL",
+    LIGHTING_SPACE.LIBRARY_READING_AREA: "UNDETERMINED",
+    LIGHTING_SPACE.LIBRARY_STACKS: "UNDETERMINED",
+    LIGHTING_SPACE.MANUFACTURING_FACILITY_DETAILED_MANUFACTURING_AREA: "ALL_OTHER",
+    LIGHTING_SPACE.MANUFACTURING_FACILITY_EQUIPMENTROOM: "ALL_OTHER",
+    LIGHTING_SPACE.MANUFACTURING_FACILITY_EXTRA_HIGH_BAY_AREA: "ALL_OTHER",
+    LIGHTING_SPACE.MANUFACTURING_FACILITY_HIGH_BAY_AREA: "ALL_OTHER",
+    LIGHTING_SPACE.MANUFACTURING_FACILITY_LOW_BAY_AREA: "ALL_OTHER",
+    LIGHTING_SPACE.MUSEUM_GENERAL_EXHIBITION_AREA: "ALL_OTHER",
+    LIGHTING_SPACE.MUSEUM_RESTORATION_ROOM: "ALL_OTHER",
+    LIGHTING_SPACE.POST_OFFICE_SORTING_AREA: "ALL_OTHER",
+    LIGHTING_SPACE.RELIGIOUS_FACILITY_FELLOWSHIP_HALL: "ALL_OTHER",
+    LIGHTING_SPACE.RELIGIOUS_FACILITY_WORSHIP_PULPIT_CHOIR_AREA: "ALL_OTHER",
+    LIGHTING_SPACE.RETAIL_FACILITIES_DRESSING_FITTING_ROOM: "RETAIL",
+    LIGHTING_SPACE.RETAIL_FACILITIES_MALL_CONCOURSE: "RETAIL",
+    LIGHTING_SPACE.SPORTS_ARENA_PLAYING_AREA_CLASS_I_FACILITY: "ALL_OTHER",
+    LIGHTING_SPACE.SPORTS_ARENA_PLAYING_AREA_CLASS_II_FACILITY: "ALL_OTHER",
+    LIGHTING_SPACE.SPORTS_ARENA_PLAYING_AREA_CLASS_III_FACILITY: "ALL_OTHER",
+    LIGHTING_SPACE.SPORTS_ARENA_PLAYING_AREA_CLASS_IV_FACILITY: "UNDETERMINED",
+    LIGHTING_SPACE.TRANSPORTATION_FACILITY_BAGGAGE_CAROUSEL_AREA: "ALL_OTHER",
+    LIGHTING_SPACE.TRANSPORTATION_FACILITY_AIRPORT_CONCOURSE: "ALL_OTHER",
+    LIGHTING_SPACE.TRANSPORTATION_FACILITY_TICKET_COUNTER: "ALL_OTHER",
+    LIGHTING_SPACE.WAREHOUSE_STORAGE_AREA_MEDIUM_TO_BULKY_PALLETIZED_ITEMS: "WAREHOUSE",
+    LIGHTING_SPACE.WAREHOUSE_STORAGE_AREA_SMALLER_HAND_CARRIED_ITEMS: "WAREHOUSE",
+}
+
+
+def get_zone_BPF_BAT(rmd: dict, zone_id: str) -> dict[str, str]:
+    """
+     Get a dictionary of the BPF_BAT and areas for a given zone.
+
+     Parameters
+    ----------
+    RMD: str, The ruleset model descriptions
+    zone_id: str, The id of the zone
+
+    Returns
+    -------
+    zone_BPF_BAT_dict: A dict for the zone that saves the BPF_BAT as keys and the areas as the values.
+                       Example: {SCHOOL: 50000, UNDETERMINED: 2000}
+
+    """
+
+    zone_BPF_BAT_dict = {}
+    for space in find_all(
```

Comment:
We realized that `find_all()` actually takes `rpd`, instead of `rmd` as the argument. So the search path must begin with `$.ruleset_model_descriptions[*]`

Reply:
Actually maybe that was only in the context of the schema validation. Do you think this is correct to pass `rmd` here?

Reply:
Disregard my comments here. 

We just need to make sure that `get_BPF_building_area_types_and_zones()` is provided with the `rmd` object when it is called, rather than the `rpd` object.

Reply:
Sounds great! Thanks for the clarification.